src/unix/stream.c

```diff
@@ -1127,7 +1127,12 @@ static void uv__read(uv_stream_t* stream) {
     assert(stream->alloc_cb != NULL);
 
     buf = uv_buf_init(NULL, 0);
+#if defined(_AIX) || defined(__linux__) || defined(__APPLE__)
+    stream->alloc_cb((uv_handle_t*)stream, stream->type == UV_TCP ?
+                    ((uv_tcp_t *) stream)->chunk_read_size : STDTCPWINDOW, &buf);
```

Comment:
@saghul this is the part you think could be handled differently to be more forward compatible to uv 2.x?